zebra-network/src/meta_addr.rs

```diff
@@ -458,6 +458,25 @@ impl MetaAddr {
         }
     }
 
+    /// Has this peer been seen recently?
+    ///
+    /// Returns `true` if this peer has responded recently or if the peer was gossiped with a
+    /// recent reported last seen time.
+    ///
+    /// [`constants::REACHABLE_PEER_DURATION`] represents the maximum time since a peer was seen to
+    /// still be considered reachable.
+    pub fn was_recently_reachable(&self) -> bool {
+        if let Some(last_seen) = self.last_seen() {
+            // Correctness: `last_seen` shouldn't ever be in the future, either because we set the
+            // time or because another peer's future time was sanitized when it was added to the
+            // address book
+            last_seen.saturating_elapsed() < constants::REACHABLE_PEER_DURATION
```

Comment:
Possible Bug: Why are we excluding peers which are ready right now?

This will stop Zebra gossiping its own listener address, and the addresses of any peers it's actively communicating with.